SAT/Z3Interfacing.cpp

```diff
@@ -49,19 +49,19 @@ struct BottomUpChildIter<z3::expr>
   z3::expr _self;
 
   /** constructs an iterator over the children of the current node */
-  BottomUpChildIter(z3::expr a) : _idx(0), _self(a) {}
+  BottomUpChildIter(z3::expr a, EmptyContext = EmptyContext()) : _idx(0), _self(a) {}
```

Comment:
Hi Joe, coming back to reading the diff for this PR.

Here, I am getting quite baffled. What is `EmptyContext` and why does a partial specialization of `BottomUpChildIter` need one in its constructor signature (and elsewhere), while also immediately ignoring it (and at the same time supplying a default argument for it)?

This might not be an issue of Z3Interfacing, but perhaps of some (arguably) missing documentation in `BottomUpEvaluation.hpp`? I am sure you hade done some cool work there, but if future vampire developers are to benefit from it, we need more "self-explanatory-dness" or perhaps more comments somewhere. (Some comments I sure might have missed - feel free to point me where appropriate ;))

Reply:
There's actually some documentation missing. I just pushed an update!